rotkehlchen/usage_analytics.py

```diff
@@ -22,134 +24,123 @@
 
 class GeolocationData(NamedTuple):
     country_code: str
-    city: str
 
 
-def query_ipwhoisio() -> Optional[GeolocationData]:
-    """10,000 requests per month per IP
-    https://ipwhois.io/documentation
-    """
+def retrieve_location_data(data_dir: Path) -> Optional[GeolocationData]:
+    """This functions tries to get the country of the user based on the ip.
+    To do that it makes use of an open ip to country database and tries to obtain
+    the ip ussing UPnP protocol."""
+    # Download a copy of database with maps of ip to country
+    geoip_dir = data_dir / 'misc'
+    geoip_dir.mkdir(parents=True, exist_ok=True)
+    # get latests database version
+    metadata_query_failed = False
     try:
         response = requests.get(
-            'http://free.ipwhois.io/json/',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
+            url='https://api.github.com/repos/geoacumen/geoacumen-country/branches/master',
+            timeout=DEFAULT_TIMEOUT_TUPLE,
         )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
-        return None
-
-    try:
-        json_ret = jsonloads_dict(response.text)
-    except JSONDecodeError:
-        return None
-
-    return GeolocationData(
-        country_code=json_ret.get('country_code', 'unknown'),
-        city=json_ret.get('city', 'unknown'),
-    )
-
-
-def query_ipinfo() -> Optional[GeolocationData]:
-    """
-    50,000 requests per month tied to the API Key
-    https://ipinfo.io/developers
-    """
-    try:
-        response = requests.get(
-            'https://ipinfo.io/json?token=16ab40aad9bd5b',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
+        data = response.json()
+        date = deserialize_timestamp_from_date(
+            date=data['commit']['commit']['author']['date'],
+            formatstr='%Y-%m-%dT%H:%M:%S',
+            location='Analytics',
         )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
-        return None
-
+    except requests.exceptions.RequestException as e:
+        log.debug(f'Failed to get metadata information for geoip file. {str(e)}')
+        metadata_query_failed = True
+    except DeserializationError as e:
+        log.debug(f'Failed to deserialize date in metadata information for geoip file. {str(e)}')
+        metadata_query_failed = True
+    except KeyError as e:
+        log.debug(f'Github response for geoip file had missing key {str(e)}')
+        metadata_query_failed = True
+
+    if metadata_query_failed:
+        old_files = list(geoip_dir.glob('*.mmdb'))
+        if len(old_files) == 0:
+            return None
+        filename = old_files[0]
+    else:
+        filename = geoip_dir / f'geoip-{date}.mmdb'
+
+    if not filename.is_file():
+        # Remove old files
+        files = geoip_dir.glob('*.*')
+        for f in files:
+            os.remove(f)
```

Comment:
You use pathlib elsewhere. Use it here too: https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink